docs/source/advanced/processors.rst

```diff
@@ -0,0 +1,120 @@
+.. _processors:
+
+Processors
+==========
+
+.. contents::
+    :local:
+    :backlinks: entry
+
+Processors are a middleware pipeline that alters the records transferred from inlets to outlets. Two most common usages of these would be:
+
+* Filtering - removing some or all records before feeding them to outlets.
+* Transforming - altering the records before feeding them to outlets.
+
+Simple example
+--------------
+
+.. code-block:: python
+
+    # Example filtering
+    def only_large_numbers(records: List[Records]):
+        result = []
+        for record in records:
+            if record.payload >= 100:
+                result.push(record)
+        return result
+
+    # Example transforming:
+    def round_to_integers(records: List[Records]):
+        for record in records:
+            record.payload = round(record.payload)
+        return records
+
+    # pass to a link
+    link = Link(..., processors=[only_large_numbers, round_to_integers])
+
+.. rst-class:: mb-s
+
+    The processor pipeline used in the above example will turn the following list:
+
+    .. rst-class:: highlight-small
+    .. code-block:: python
+
+        [99.999, 200, 333.333]
+
+.. rst-class:: mb-s
+
+    into:
+
+    .. rst-class:: highlight-small
+    .. code-block:: python
+
+        [200, 333]
+
+Note that :code:`99.999` got filtered out given the order of processors. If we were to swap the processors, the rounding would occur before filtering, allowing all three results through the filter.
+
+Processors explained
+--------------------
+
+A processor is a :any:`callable` function that accepts a list of records and returns a (potentially altered) list of records.
+
+Processors are called in the order in which they are passed, after all inlets finish producing their data. The result of each processor is given to the next one, until finally the resulting records continue the transfer normally.
+
+.. _link-outlet-processors:
+
+Link vs Outlet processors
+--------------------------
+
+Databay supports two types of processors, depending on the scope at which they operate. This distinction can be used to determine at which level a particular processor is to be applied.
+
+* :any:`Link processor <Link>` - passed to links and applied to all records transferred by that link.
```

Comment:
This is the only part of this documentation section I find myself rereading through a bit of confusion. I think being a bit more explicit on the level at which the processor is operating might be good. 

For example I'd say `Link processor <Link> - passed to links and applied to all records transferred from the inlet(s) to that link.` I'd again say something similar for the Outlet processor. 

Another thing is you've done some really nice work on the basic graph images showing how inlets connect outlets. I think the impact of doing a graph like the one at `databay/build/_images/databay_basic_graph.png` for these processors would be huge as it'd give an immediate visual indication of where the processor functionality kicks in depending on what the user chooses.

Reply:
Ok gotcha man, I'll rework this section.

Great suggestion on doing some more graphs to explain this! 👏